app/src/main/java/it/niedermann/nextcloud/deck/persistence/sync/SyncManager.java

```diff
@@ -166,76 +172,130 @@ public void onError(Throwable throwable) {
 
     @AnyThread
     public void synchronize(@NonNull IResponseCallback<Boolean> responseCallback) {
-        if(responseCallback.getAccount() == null) {
+        synchronize(Collections.singletonList(responseCallback));
+    }
+    private void synchronize(@NonNull List<IResponseCallback<Boolean>> responseCallbacks) {
+        IResponseCallback responseCallback = responseCallbacks.get(0);
+        Account callbackAccount = responseCallback.getAccount();
+        if(callbackAccount == null) {
             throw new IllegalArgumentException(Account.class.getSimpleName() + " object in given " + IResponseCallback.class.getSimpleName() + " must not be null.");
         }
-        if(responseCallback.getAccount().getId() == null) {
+        Long callbackAccountId = callbackAccount.getId();
+        if(callbackAccountId == null) {
             throw new IllegalArgumentException(Account.class.getSimpleName() + " object in given " + IResponseCallback.class.getSimpleName() + " must contain a valid id, but given id was null.");
         }
-        doAsync(() -> refreshCapabilities(new IResponseCallback<Capabilities>(responseCallback.getAccount()) {
-            @Override
-            public void onResponse(Capabilities response) {
-                if (!response.isMaintenanceEnabled()) {
-                    if (response.getDeckVersion().isSupported(appContext)) {
-                        long accountId = responseCallback.getAccount().getId();
-                        Date lastSyncDate = LastSyncUtil.getLastSyncDate(responseCallback.getAccount().getId());
-                        Date now = DateUtil.nowInGMT();
+        List<IResponseCallback<Boolean>> queuedCallbacks = RUNNING_SYNCS.get(callbackAccountId);
+        if (queuedCallbacks != null){
+            queuedCallbacks.addAll(responseCallbacks);
+            return;
+        } else {
+            queuedCallbacks = new ArrayList<>();
+            queuedCallbacks.addAll(responseCallbacks);
+            RUNNING_SYNCS.put(callbackAccountId, queuedCallbacks);
+        }
+        doAsync(() -> {
+            List<IResponseCallback<Boolean>> callbacksQueueForSync = new ArrayList<>();
+            for (IResponseCallback<Boolean> cb : RUNNING_SYNCS.get(callbackAccountId)) {
+                callbacksQueueForSync.add(cb);
+            }
```

Comment:
`List<IResponseCallback<Boolean>> callbacksQueueForSync = new ArrayList<>(RUNNING_SYNCS.get(callbackAccountId));`

Reply:
Nope, won't happen. Not even in race-conditions. The callbacks are added right before the async task starts, and are only removed afterwards. No chance for a NPE here.

Reply:
Well, no chance *yet* - It is likely that i have no idea about this stuff in 4 months and if i change something in the SyncManager it might break in the future... That's what i mean with defensice coding: Defend the backend from me.

Reply:
Point taken, will be fixed tomorrow.

Reply:
done.